components/ai_chat/core/browser/utils.cc

```diff
@@ -178,6 +183,65 @@ const std::string& GetActionTypeQuestion(mojom::ActionType action_type) {
   return iter->second;
 }
 
+uint32_t GetMaxPageContentLength(std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> result =
+      context_size.value_or(kCustomModelDefaultContextSize) *
+      kDefaultCharsPerToken;
+  if (!result.IsValid()) {
+    return std::numeric_limits<uint32_t>::max();
+  }
+  return result.ValueOrDie();
+}
+
+uint32_t GetMaxPageContentLength(const mojom::Model& model) {
+  // For Leo models, return the max_page_content_length directly.
+  if (model.options->is_leo_model_options()) {
+    return model.options->get_leo_model_options()->max_page_content_length;
+  }
+
+  // For custom models, delegate to the context size version.
+  return GetMaxPageContentLength(
+      model.options->get_custom_model_options()->context_size);
+}
+
+uint32_t GetLongConversationWarningLength(
+    std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> checked_context_size =
+      context_size.value_or(kCustomModelDefaultContextSize);
+  base::CheckedNumeric<uint32_t> result =
+      checked_context_size * kDefaultCharsPerToken * kMaxContentLengthThreshold;
+  if (!result.IsValid()) {
+    return std::numeric_limits<uint32_t>::max();
+  }
+  return result.ValueOrDie();
+}
+
+uint32_t GetLongConversationWarningLength(const mojom::Model& model) {
+  // For Leo models, return the warning limit directly.
+  if (model.options->is_leo_model_options()) {
+    return model.options->get_leo_model_options()
+        ->long_conversation_warning_character_limit;
+  }
+
+  // For custom models, delegate to the context size version.
+  return GetLongConversationWarningLength(
+      model.options->get_custom_model_options()->context_size);
+}
+
+bool IsValidContextSize(const std::optional<uint32_t>& context_size) {
+  const uint32_t MIN_CONTEXT_SIZE = 1;
+  const uint32_t MAX_CONTEXT_SIZE = std::numeric_limits<int32_t>::max();
+
+  if (!context_size) {
+    return true;  // Allow nullopt (no value provided)
+  }
+
+  base::CheckedNumeric<uint32_t> size = *context_size;
```

Comment:
You don't need check numerics here... there's no conversion or arithmetics taking place.